src/main/java/org/springframework/data/couchbase/repository/query/CouchbaseQueryMethod.java

```diff
@@ -182,6 +185,24 @@ public String getInlineN1qlQuery() {
 		return StringUtils.hasText(query) ? query : null;
 	}
 
+	/**
+	 * is this a 'delete'?
+	 *
+	 * @return is this a 'delete'?
+	 */
+	public boolean isDeleteQuery() {
+		return getName().toLowerCase().startsWith("delete");
+	}
+
+	/**
+	 * is this an 'exists' query?
+	 *
+	 * @return is this an 'exists' query?
+	 */
+	public boolean isExistsQuery() {
+		return getName().toLowerCase().startsWith("exists");
```

Comment:
should we use Locale.ROOT in the toLowerCase?

Reply:
done